File: fren/app/src/main/java/com/gtek/fren/MainActivity.java
package com.gtek.fren;

import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import com.google.android.material.navigation.NavigationView;

import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.appcompat.app.AppCompatActivity;

import com.gtek.fren.databinding.ActivityMainBinding;
import com.gtek.fren.ui.helper.EmotionBenchmark;
import com.gtek.fren.ui.helper.EmotionClassifier;

import java.io.IOException;

public class MainActivity extends AppCompatActivity {
    private AppBarConfiguration mAppBarConfiguration;
    private EmotionClassifier emotionClassifier;
    private EmotionBenchmark emotionBenchmark;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        com.gtek.fren.databinding.ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setSupportActionBar(binding.appBarMain.toolbar);

        DrawerLayout drawer = binding.drawerLayout;
        NavigationView navigationView = binding.navView;

        // Inisialisasi EmotionClassifier
        emotionClassifier = new EmotionClassifier(this);
        emotionBenchmark = new EmotionBenchmark();

        mAppBarConfiguration = new AppBarConfiguration.Builder(
                R.id.nav_emotion_analysis, R.id.nav_about, R.id.nav_about_emotion, R.id.nav_privacy_policy)
                .setOpenableLayout(drawer)
                .build();
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_main);
        NavigationUI.setupActionBarWithNavController(this, navController, mAppBarConfiguration);
        NavigationUI.setupWithNavController(navigationView, navController);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == R.id.model_cnnresnet || id == R.id.model_kanresnet) {
            try {
                if (id == R.id.model_cnnresnet) {
                    emotionClassifier.switchModel(EmotionClassifier.MODEL_CNN_RESNET);
                } else {
                    emotionClassifier.switchModel(EmotionClassifier.MODEL_KAN_RESNET);
                }

                item.setChecked(true);

                String modelName = (id == R.id.model_cnnresnet) ? "CNN ResEmoteNet" : "KAN ResEmoteNet";
                Toast.makeText(this, "Switched to " + modelName, Toast.LENGTH_SHORT).show();

                invalidateOptionsMenu();

                if (emotionBenchmark != null) {
                    emotionBenchmark.reset();
                }

                //TODO
                //Clear resultList when uploading a new image

                return true;
            } catch (IOException e) {
                Log.e("MainActivity", "Error switching model: " + e.getMessage());
                Toast.makeText(this, "Error switching model", Toast.LENGTH_SHORT).show();
            }
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public boolean onPrepareOptionsMenu(Menu menu) {
        MenuItem selectModelItem = menu.findItem(R.id.action_select_model);
        if (selectModelItem != null && selectModelItem.hasSubMenu()) {
            Menu subMenu = selectModelItem.getSubMenu();
            for (int i = 0; i < subMenu.size(); i++) {
                MenuItem subMenuItem = subMenu.getItem(i);
                if (subMenuItem.getItemId() == R.id.model_cnnresnet) {
                    subMenuItem.setChecked(emotionClassifier.getCurrentModel().equals(EmotionClassifier.MODEL_CNN_RESNET));
                } else if (subMenuItem.getItemId() == R.id.model_kanresnet) {
                    subMenuItem.setChecked(emotionClassifier.getCurrentModel().equals(EmotionClassifier.MODEL_KAN_RESNET));
                }
            }
        }
        return super.onPrepareOptionsMenu(menu);
    }

    @Override
    public boolean onSupportNavigateUp() {
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_main);
        return NavigationUI.navigateUp(navController, mAppBarConfiguration)
                || super.onSupportNavigateUp();
    }
}

File: fren/app/src/main/java/com/gtek/fren/MainActivity.kt
package com.gtek.fren

import android.os.Bundle
import android.util.Log
import android.view.Menu
import com.google.android.material.navigation.NavigationView
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import androidx.drawerlayout.widget.DrawerLayout
import androidx.appcompat.app.AppCompatActivity
import com.gtek.fren.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var appBarConfiguration: AppBarConfiguration
    private lateinit var binding: ActivityMainBinding
    private val TAG = "MainActivity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(TAG, "onCreate executed")

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        Log.d(TAG, "View binding initialized")

        setSupportActionBar(binding.appBarMain.toolbar)
        Log.d(TAG, "Toolbar set as support action bar")

        val drawerLayout: DrawerLayout = binding.drawerLayout
        val navView: NavigationView = binding.navView
        val navController = findNavController(R.id.nav_host_fragment_content_main)

        // Passing each menu ID as a set of IDs because each menu should be considered as top-level destinations.
        appBarConfiguration = AppBarConfiguration(
            setOf(R.id.nav_home, R.id.nav_gallery, R.id.nav_slideshow),
            drawerLayout
        )
        setupActionBarWithNavController(navController, appBarConfiguration)
        navView.setupWithNavController(navController)

        Log.d(TAG, "Navigation components set up")
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        Log.d(TAG, "onCreateOptionsMenu called")
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onSupportNavigateUp(): Boolean {
        Log.d(TAG, "onSupportNavigateUp called")
        val navController = findNavController(R.id.nav_host_fragment_content_main)
        val navigateUp = navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
        Log.d(TAG, "Navigate up result: $navigateUp")
        return navigateUp
    }
}
